File: snippets/kotlin_mpp_lib/shared/src/commonMain/kotlin/com/example/kotlinmpplib/ServiceStatus.kt
package com.example.kotlinmpplib

import breez_sdk.*
class ServiceStatus {
    fun healthCheckStatus(sdk: BlockingBreezServices) {
        // ANCHOR: health-check-status
        try {
            val healthCheck = serviceHealthCheck("<api key>")
            // Log.v("Breez", "Current service status is: ${healthCheck.status}")
        } catch (e: Exception) {
            // Handle error
        }
        // ANCHOR_END: health-check-status
    }

    fun reportPaymentFailure(sdk: BlockingBreezServices) {
        // ANCHOR: report-payment-failure
        val paymentHash = "..."
        try {
            sdk.reportIssue(ReportIssueRequest.PaymentFailure(
                ReportPaymentFailureDetails(paymentHash)))
        } catch (e: Exception) {
            // Handle error
        }
        // ANCHOR_END: report-payment-failure
    }
}

File: snippets/react-native/service_status.ts
import { serviceHealthCheck, reportIssue, ReportIssueRequestVariant } from '@breeztech/react-native-breez-sdk'

const healthCheckStatus = async () => {
  // ANCHOR: health-check-status
  try {
    const healthCheck = await serviceHealthCheck('<api key>')
    console.log(`Current service status is: ${healthCheck.status}`)
  } catch (err) {
    console.error(err)
  }
  // ANCHOR_END: health-check-status
}

const reportPaymentFailure = async () => {
  // ANCHOR: report-payment-failure
  try {
    const paymentHash = '...'
    await reportIssue({
      type: ReportIssueRequestVariant.PAYMENT_FAILURE,
      data: { paymentHash }
    })
  } catch (err) {
    console.error(err)
  }
  // ANCHOR_END: report-payment-failure
}
